src/periodic.jl

```diff
@@ -0,0 +1,93 @@
+"""
+    Periodic{T}(f, period::T, phase_shift::T) <: Transform
+
+Applies a periodic function `f` with provided `period` and `phase_shift` to the data.
+
+!!! note
+    For `TimeType` data, the result will change depending on the type of `period` given,
+    even if the same amount of time is described. Example: `Week(1)` vs `Second(Week(1))`;
+    the former starts the period on the most recent Monday, while the latter starts the
+    period on the most recent multiple of 604800 seconds since time 0.
+
+# Fields
+* `f::Union{typeof(cos), typeof(sin)}`: the periodic function
+* `period::Union{Real, Period}`: the function period. Must be strictly positive.
+* `phase_shift::Union{Real, Period}`: adjusts the phase of the periodic function, measured
+    in the same units as the input. Increasing the value translates the function to the
+    right, toward higher/later input values.
+"""
+struct Periodic{T} <: Transform where T <: Union{Real, Period}
+    f::Union{typeof(cos), typeof(sin)}
+    period::T
+    phase_shift::T
+
+    function Periodic(f, period::T, phase_shift::T) where T
+        period <= zero(T) && throw(ArgumentError("period must be strictly positive."))
```

Comment:
```suggestion
        period > zero(T) || throw(ArgumentError("period must be strictly positive."))
```

I realise this is equivalent but perhaps uses fewer characters.

Reply:
It's cool I like how it reads as "this had better be positive OR I'll complain" 😄